pika/adapters/tornado_connection.py

```diff
@@ -72,7 +73,10 @@ def _adapter_disconnect(self):
         """Disconnect from the RabbitMQ broker"""
         if self.socket:
             self.ioloop.remove_handler(self.socket.fileno())
-        super(TornadoConnection, self)._adapter_disconnect()
+        try:
+            super(TornadoConnection, self)._adapter_disconnect()
+        except exceptions.AMQPConnectionError as e:
```

Comment:
I used ```on_open_error_callback``` because at that point, the state of the connection is not in ```CONNECTION_OPEN``` yet. Also, I think you do not get a reason/message from the broker. The connection essentially times out. That's why ```ProbableAuthenticationError```.

I leave this decision to @gmr .